### PR TITLE
feat: add type definition 

### PR DESCRIPTION
**Pull request by harisarang**
## PR Checklist

- [ ] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+type SearchClient = object;
+
+interface TypesenseNode {
+  host: string;
+  port: string;
```
Comment (TheMartes) [inline index.d.ts:5]: Port should be always a number. You can convert it to string whenever you want and when it makes sense, or if you want to use it in connection string. But from the networking point of view, port is always a number and it should be handled like that.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  protocol: string;
```
Comment (TheMartes) [inline index.d.ts:6]: And Protocol should be always a choice between HTTP / HTTPS. You can make a enum for example something like `supported-protocols.enum.ts` where there will be all the protocol listed and allows only protocols which are listed there. :) People can quickly see in intelli sense what protocols are supported!

**Review by TheMartes — COMMENTED** [review]
Just a few things on Node Configuration. :)

File: index.d.ts (hunk 1 above)
Comment (harisarang) [reply index.d.ts:5]: Yeah true actually but in the constructor it is expecting a `string`.

File: index.d.ts (hunk 2 above)
Comment (harisarang) [reply index.d.ts]: Fixed that !

File: index.d.ts (hunk 1 above)
Comment (TheMartes) [reply index.d.ts:5]: Well then it's okay :)

**Pull request merged**